Add activity report for Q1 2023 and roadmap for Q2. (#283)
scala/scala.epfl.ch: records/2023-Q1-roadmap.md

---
layout: contact
---

> This page lists the projects that the Scala Center will work on during

Reviewer: I think we should use a present tense with `the current quarter`

```suggestion
This page lists the projects that the Scala Center are working on during
```

Author: I am not sure, maybe @ckipp01 could help us? :pray:

Reviewer 2: Yup, I'd also recommend using `are working` since it's present tense.

the current quarter. We also post regular updates about our projects on the
[Scala Contributors forum](https://contributors.scala-lang.org/c/scala-center/25).

To have more information about our _completed_ projects, please see the
[quarterly activity reports]({% link records.md %}).

## Roadmap for 2023 Q1
{: .no_toc}

The following sections present our plan for the current quarter. Every

> project description is followed by the concrete results we will deliver, and
> their expected outcome on the Scala community.

Reviewer 2: ```suggestion
project description is followed by the concrete results we will deliver and
their expected outcome on the Scala community.
```

* Table of Contents
  {:toc}

### Make Scala tooling seamless for newcomers, and provide robust foundations for the Scala ecosystem

#### Polish sbt and Coursier

> Most of the Scala projects are built with sbt, which uses Coursier under the hood.

Reviewer 2: ```suggestion
Most Scala projects are built with sbt, which uses Coursier under the hood.
```

library to read TASTy files, explore them, and ask semantic questions about
them. It is independent from the compiler, and can be used as a building
block to implement tools performing static analyses of Scala programs. An
example is TASTY-MiMa, an upcoming tool “like
[MiMa](https://github.com/lightbend/mima), but for TASTy”.

We plan to implement support for semantic queries such as “are two types
equivalent?”, “is a type a subtype of another one?”, etc., and to publish a
prototype of TASTy-MiMa.

You can read the full roadmap
[here](https://contributors.scala-lang.org/t/the-scala-centers-roadmap-for-tasty-query/5564).

#### Create a Stable API for the Scala 3 Presentation Compiler

Currently, Metals works only for a limited subset of Scala 3 versions. Furthermore, we need to publish
a new release of Metals after every new release of the compiler to support it.

We would like to create a stable API for the Scala 3 presentation compiler to untie Metals to the compiler
release cycle and support a wider range of Scala 3 versions (including experimental versions of Scala 3).

#### Reduce the Number of Ways to Import Scala Projects in IDEs

> Currently, when starting new projects in Metals the build server that is used by default is Bloop. Part of this is for historical reasons, but part of it is also due to the speed of Bloop and the integrations that exist for it. Bloop has been incredibly important in the Build Server arena. However, there is also BSP implementation in sbt and in Mill and even when a `.bsp/<build-tool>.json` entry exists, Metals will still force Bloop on the user unless they manually switch. This can cause confusion and also may add extra steps into choosing a build server that shouldn't be needed.

Reviewer 2: ```suggestion
Currently, when starting new projects in Metals the build server that is used by default is Bloop. Part of this is for historical reasons, but part of it is also due to the speed of Bloop and the integrations that exist for it. Bloop has been incredibly important in the Build Server arena. However, there are also BSP implementations in sbt and in Mill and even when a `.bsp/<build-tool>.json` entry exists, Metals will still force Bloop on the user unless they manually switch. This can cause confusion and also may add extra steps in choosing a build server that shouldn't be needed.
```

This problem also exists in a different way in IntelliJ when you open for example an sbt project that has an existing `.bsp/sbt.json` file you'll be prompted to import the project either as a BSP project or an sbt project. Arguably, newcomers have no idea what BSP is, and they shouldn't have to.

We'd like to minimize the number of ways to import projects and revisit the default choices
that Metals makes of using Bloop no matter the build tool. The aim is to make
the experience a newcomer has smoother, abstracting away things like Bloop and
BSP, while still allowing for a powerful user experience for advanced users. Read the full roadmap and progress
reports [here](https://contributors.scala-lang.org/t/revisiting-the-default-build-server-for-metals-roadmap/6054).

### Education and Documentation

#### Scala Website

We plan to modernize the Scala website, and to integrate better the Scala 2
and Scala 3 documentation.

Following up on [the work]({% link records/2022-Q4-activity-report.md %})
we did in the previous quarter, we plan to polish further the content of the
website.

The next main tasks are:

> - make the content more use-case-oriented rather than features-oriented,
> - reorganize the structure to make it more intuitive according to the reader’s background,
> - create new tutorials to document better how to contribute to the Scala open-source
>   ecosystem.

Reviewer 2: ```suggestion
- make the content more use-case-oriented rather than features-oriented
- reorganize the structure to make it more intuitive according to the reader’s background
- create new tutorials to document better how to contribute to the Scala open-source
  ecosystem
```
